### DISCUSSION TITLE
How to understand the base value of SHAP?

### DISCUSSION THREAD

**Discussion by LY2260789** [shap/shap#3831] [Q&A]
As we know, SHAP interpreted the model output (the predicted value) as the sum of the global expected value (the base value) and the marginal contribution of each feature (the Shapley value of each feature). Why is this relationship, and how should we understand the global expected value of the model output (the base value)?

**Comment by Barszo** [comment]
This is a very good question. I have been wondering about the same thing for some time now. As far as I understand, Shapley values do not tell you how important a feature is **in general**, but rather how important it is for **shifting the prediction from the base value towards the given prediction**. This makes Shapley value estimates somewhat 'unstable', as you can read in this article: https://arxiv.org/abs/2204.11351.

The basic idea is as follows: you have a model $f$, one sample of features $\boldsymbol{x}_e$, and the prediction $f(\boldsymbol{x}_e) = y_e$. Now, you want to compare this with the base value $y_b$, which is obtained by putting the mean features into the model: $f(\overline{\boldsymbol{x}}) = y_b$. What SHAP tells you is which feature, and by how much, moved the prediction from $y_b$ towards $y_e$.

The fluctuations mentioned earlier arise from changing the value of $y_b$. For example, you can take 100 samples from your dataset to calculate the base value $y'_b$, and then take a different set of 100 samples from the dataset to calculate a different base value $y''_b$. The Shapley values might differ depending on whether you use $y'_b$ or $y''_b$.

As for the base value itself, we use the most probable output of the model because (in my opinion) it can give us the most insight into the general behaviour of the model. It tells us which features caused the change from the most probable $y_b$ towards another prediction $y_e$. However, if you’d like to know specifically why the model changed the prediction from a different $y''_b$ to your prediction $y_e$, I don't see any reason not to adjust the base value to better suit your needs.

While SHAP is not very useful for allowing custom base values out-of-the-box, it could be modified to some extent.
